mono/mini/main.c

```diff
@@ -179,8 +182,65 @@ probe_embedded (const char *program, int *ref_argc, char **ref_argv [])
 		goto doclose;
 	if (read (fd, sigbuffer, sizeof (sigbuffer)) == -1)
 		goto doclose;
-	if (memcmp (sigbuffer+sizeof(uint64_t), "xmonkeysloveplay", 16) != 0)
+	// First, see if "xmonkeysloveplay" is at the end of file
+	if (memcmp (sigbuffer + sizeof (uint64_t), "xmonkeysloveplay", 16) == 0)
+		goto found;
+	else
+	{
+#ifdef HOST_DARWIN
```

Comment:
We only want this on MacOS, not iOS etc right? This define should be `TARGET_OSX`, sorry for not catching that earlier.

Reply:
That's what HOST_DARWIN does!

Reply:
Oh, I don't know about ipad / iphone / ios - do they also have a signing process?

Reply:
`HOST_DARWIN` covers all the Apple platforms: MacOS, iOS, tvOS, etc. The entrypoint for the mobile platforms is different and mkbundle doesn't exist for them, so this should only be on MacOS, and accordingly should use the `TARGET_OSX` define. I realize that's a switch from host to target, but that's an issue on our end with the define names (sorry!) and `TARGET_OSX` is what you want.

Reply:
Okay, done - https://github.com/mono/mono/commit/b1ca14f06aed4e1683a6e8cf04db4c2ddd502ea1 - the collection of HOST_* and TARGET_* are quite confusing, especially as WIN32/WIN64 exist for both! I know there are subtle differences regarding cross-compiling, nonetheless...

I think the new code is harmless on the other apple OSes, except for a small waste of cpu cycles.

Reply:
Yeah, for our AOT compiler cross-compiling is a supported scenario, so the HOST/TARGET difference matters in some cases. We should really fix up the OSX defines, though it's not particularly urgent.